mne/io/tests/test_reference.py

```diff
@@ -473,4 +474,41 @@ def test_add_reference():
     pytest.raises(ValueError, add_reference_channels, raw, 1)
 
 
+def test_bipolar_combinations():
+    """Test bipolar channel generation."""
+    channels = ['CH' + str(ni + 1) for ni in range(10)]
+    info = create_info(
+        ch_names=channels, sfreq=1000., ch_types=['eeg'] * len(channels))
+    raw_data = np.random.randn(len(channels), 1000)
+    raw = RawArray(raw_data, info)
+
+    def _check_bipolar(raw_test, ch_a, ch_b):
+        assert_array_equal(
+            raw_test.get_data(
+                picks=[raw_test.ch_names.index(ch_a + '-' + ch_b)])[0, :],
+            raw_data[channels.index(ch_a), :] -
+            raw_data[channels.index(ch_b), :])
+
+    # test classic EOG/ECG bipolar reference (only two channels per pair).
+    raw_test = set_bipolar_reference(raw, ['CH2'], ['CH1'], copy=True)
+    _check_bipolar(raw_test, 'CH2', 'CH1')
+
+    # test all combinations.
+    bipolars = list(itertools.combinations(channels, 2))
+    a_channels, b_channels = [], []
+    for ch_a, ch_b in bipolars:
+        a_channels.append(ch_a)
+        b_channels.append(ch_b)
```

Comment:
`a_channels, b_channels =*zip(itertools.combinations(channels, 2))`?

Reply:
The closest expression using zip would be this one (not 100% sure):
```
a_channels, b_channels = [
        list(c) for c in zip(*itertools.combinations(channels, 2))
]
```
...is not as direct as the original, which one would be better?

Reply:
Ah yes, I mixed up the order of `zip` and the star ...

```
a_chans, b_chans = zip(*itertools.combinations(channels, 2))
raw_test = set_bipolar_reference(raw, list(a_channels), list(b_channels), copy=True)
```

Right?

Reply:
As we need the channels to compare the results after `set_bipolar_reference`, the other alternative is unroll the for (otherwise the generator code should be repeated): 
```
a_channels, b_channels = zip(*itertools.combinations(channels, 2))
a_channels = list(a_channels)
b_channels = list(b_channels)
```
I think this would be the best.